### ISSUE TITLE
[Feature Request] Add a Toggle for "Target=Enemy==Combat"

### ISSUE THREAD

**Issue by Arkios64** [PhenomDevel/BlizzHUDTweaks#3]
Basically, it's really not necessary to go full combat mode when turning in a quest or interacting with other NPCs.
The basic idea is, to distinguish between the targeted entity being friendly or not for applying the Override.

Alternatively, or additionally, being able to set "Target==Combat" individually would also work wonders.

It's currently a hard choice between having to mouseover everything before combat to check the cooldowns or revving up the full killing engine every time you access the Auction House.

**Comment by PhenomDevel** [comment]
Just so I understand it correctly.

What you mean is that you want to be able to set `Treat target as in combat` for each bar/frame individually.

Or alternatively be able to set for what type of targets the option should be active like `Friendly, Enemy, Both` ?

**Comment by PhenomDevel** [comment]
What comes to my mind when thinking about it is something like:
![image](https://user-images.githubusercontent.com/13941819/203432593-5925ee38-bcd1-48bc-b6d0-e7e5f61f1326.png)

So first of all you could check `Treat target as in combat` for each actionbar/frame individually. Secondly you can choose if it should be applied for friendly targets, harmful targets or both.

If a target is harmful is checked by `UnitCanAttack` API.

**Comment by Arkios64** [comment]
Yes, exactly.
Although, I think it might be enough to be able to set this globally and only for both/hostile, but it still might find some applications with the FULL functionality.

The main reason you might want to set it individually would be, to maybe only show a healing/buff bar when targeting a friendly unit and otherwise use the other options.
Basically just extreme edge cases that aren't critical in any way.

In short, you understood me perfectly, but I think adding ONLY a distinction for the global setting should be enough.

**Comment by PhenomDevel** [comment]
With this in mind, I'd prefer it to be global only for now to keep it somewhat simple. My concern is that the options you have might be overwhelming to a point where I have to explain every little bit.

I think the select box will contain `Friendly`, `Hostile`, and `Both` just to make it clear what `Both` means. Could be a little too less in the description box. Also, `Both` will be the default, so the add-on behaves exactly the same as in the previous versions.

**Comment by PhenomDevel** [comment]
Version `1.19.1-beta` includes this feature. Let me know if it works like you expected.

**Comment by Arkios64** [comment]
The only case I see so far that would benefit from being able to be set individually would be the `TargetFrame` itself; though that is also just a case of wanting to set different percentages from the Global preset, so that might be the case for all frames, depending on the user. Which, of course, would mean setting the option for everything, complicating everything.

An edge case I just thought of would be neutral targets. Also, the enemy faction with WM-off.
Tested them out, `neutral=enemy` and `WM-off=friendly` is perfect for the purposes of starting combat or not.

So, all in all it looks good. 
Maybe `treat only living enemies as in combat` might be a useful fourth option, that is the last thing I could currently think of.
